package/ios/RNSkia-iOS/SkiaMetalSurfaceFactory.h

```diff
@@ -0,0 +1,31 @@
+#import <MetalKit/MetalKit.h>
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+
+#import "SkCanvas.h"
+#import <include/gpu/GrDirectContext.h>
+
+#pragma clang diagnostic pop
+
+using SkiaMetalContext = struct SkiaMetalContext {
+  id<MTLCommandQueue> commandQueue = nullptr;
+  sk_sp<GrDirectContext> skContext = nullptr;
+};
+
+class ThreadContextHolder {
+public:
+  static thread_local SkiaMetalContext ThreadSkiaMetalContext;
+};
+
+class SkiaMetalSurfaceFactory {
+private:
```

Comment:
This looks really good now - could we move private under public? We usually sort public - protected - private so that it is easier to see the public interface at first glance.